lib/select-builder.js

```diff
@@ -1,89 +1,74 @@
 'use strict';
 
-const { iter } = require('@metarhia/common');
-
 const { QueryBuilder } = require('./query-builder');
 
-const allowedConditions = new Set(['=', '<>', '<', '<=', '>', '>=', 'LIKE']);
-
-const supportedOps = {
-  select: Set,
-  innerJoin: Array,
-  selectDistinct: null,
-  count: Array,
-  avg: Array,
-  min: Array,
-  max: Array,
-  sum: Array,
-  where: Array,
-  groupBy: Set,
-  orderBy: Set,
-  from: Set,
-  limit: null,
-  offset: null,
-};
+const compareOperators = ['=', '<>', '<', '<=', '>', '>=', 'LIKE'];
+const setOperators = ['= ANY', 'IN', 'NOT IN'];
+const operators = [...compareOperators, ...setOperators];
 
-const functionHandlers = {
-  count: op => `count(${op.field})`,
-  avg: op => `avg(${op.field})`,
-  min: op => `min(${op.field})`,
-  max: op => `max(${op.field})`,
-  sum: op => `sum(${op.field})`,
-};
+const sqlFunctions = ['count', 'avg', 'min', 'max', 'sum'];
 
-const parseCondition = cond => {
-  if (cond === '!=') return '<>';
-  if (!allowedConditions.has(cond)) {
-    throw new Error(`The operator "${cond}" is not permitted`);
+const parseOperator = op => {
+  if (op === '!=') return '<>';
+  if (!operators.includes(op)) {
```

Comment:
I don't see a reason to change this to an Array instead of using Set, it won't have any performance benefits and is less correct from a theoretical perspective. Can you clarify this change?

Reply:
Sets have no certain order, so we need different technique to have AND/OR

Reply:
@tshemsedinov Sets have an order according to the ECMAScript spec: https://tc39.es/ecma262/#sec-set-iterable
You can see that it uses `set.[[SetData]]` internal slot, which is a List, and appends the value to the end of it when [Set#add()](https://tc39.es/ecma262/#sec-set.prototype.add) is called.